lib/mwlib/src/MW/Media/Image/Imagick.php

```diff
@@ -100,8 +100,14 @@ public function save( $filename = null, $mimetype = null )
 	 */
 	public function scale( $width, $height, $fit = true )
 	{
-		if( $fit === true )
-		{
+		if( $fit === false && $width && $height ) {
+			try {
+				$this->image->cropThumbnailImage( $width, $height );
+				$this->image->setImagePage(0, 0, 0, 0); // see https://www.php.net/manual/en/imagick.cropthumbnailimage.php#106710
+			} catch( \Exception $e ) {
+				throw new \Aimeos\MW\Media\Exception( $e->getMessage() );
+			}
+		} else {
```

Comment:
It must be `elseif( $fit == true )`, the `resizeImage()` method must be out of the if/elseif contruct and the first `if()` must contain a `return` to resample the same behavior as before.

Reply:
Sorry, I don't get your point. Especially if I evaluate only with `==`, `$fit` is either false or not (`else`). In the first case, `Imagick::cropThumbnailImage()` will be used, in the second `Imagick::resizeImage()`. The return is at the very bottom.

Reply:
If `$fit == false` and only width or height value is given, the image should be passed to `resizeImage()` without check if image fits in. This will scale up the image proportionally to the passed size. Then, the behavior is exactly the same like before your changes in that case